test/jdk/sun/tools/jcmd/TestJcmdNoAgentLoad.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import static jdk.test.lib.Asserts.*;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.List;
+
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.Utils;
+
+/*
+ * @test
+ * @bug 8304438
+ * @summary JVMTI.agent_load should obey EnableDynamicAgentLoading
+ *
+ * @library /test/lib
+ *
+ * @run main/othervm -XX:-EnableDynamicAgentLoading TestJcmdNoAgentLoad
```

Comment:
The test verifies only one scenario. I think we need to test 3 scenarios:
- -XX:-EnableDynamicAgentLoading
- +XX:-EnableDynamicAgentLoading
- default (no EnableDynamicAgentLoading arg)

Reply:
I didn't find tests for EnableDynamicAgentLoading for attach API too (to ensure this fix doesn't break it)

Reply:
Good suggestions, thanks.
Added three modes: default, disabled and enabled.
Also, added a real java agent to check if it has been loaded.
The jcmd is using the dynamic attach to load the agents, so it has to be tested as well.
Alan recently added new tests for this as well.